catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/resource/download/ReliableResourceDownloader.java

```diff
@@ -583,6 +587,7 @@ private void cleanupAfterDownload(ReliableResourceStatus reliableResourceStatus)
         // successful downloads since client reading from this InputStream will lag when
         // Callable finishes reading product's InputStream
       }
+      downloadStatusInfo.removeDownloadInfo(downloadIdentifier);
```

Comment:
❗ The download info is added during `setupDownload` but `reliableResourceStatus` will be null until a future returns in `run`. So I think download status could still slowly leak if there is an exception before the future can return to set `reliableResourceStatus` to something other than null.

We might want to see if there is a safe way to remove the download info from `downloadStatusInfo` even when `reliableResourceStatus` is null. Currently the `removeDownloadInfo` method will just call remove on two of its internal hashmaps which should be safe to call even if the given `downloadIdentifier` is not present.

Reply:
Updated. Moved the call outside of the if statement.